c++/src/kj/common.h

```diff
@@ -102,6 +102,7 @@ KJ_BEGIN_HEADER
 #include <cstring>
 #include <initializer_list>
 #include <string.h>
+#include <compare>
```

Comment:
Do we really have to include this? This header includes dozens of other headers transitively, and including it here means adding all those file to every single translation unit in our entire codebase.

If this is an absolute requirement to implement the spaceship operator, I would honestly rather implement `operator<` and `operator==` instead (which I believe the compiler can leverage to implement all the other operators).

Reply:
Because we are using such a small subset of the compare header, we could write our own version and this would turn each of the comparator functions into clean one liners instead of default. however using explicit implementations for `operator<` and `operator==` to implement `operator<=` would call `memcmp` twice on the minimum size of the arrays being compared which I think is too expensive.

Reply:
Also from exploring the implementation I've seen it only includes about 10 extra headers of which only 4 have not been included anywhere else is that really too many? Most of those headers are just macros for alignment sizes or assignments for template meta-programming objects.

The specific headers I saw were:
- `ordering.h`: the original header included mostly everything is constexpr
- `enable_if.h`: the enable_if meta-programming header everything is constexpr
- `is_same.h`: the is_same meta-programming header everything is constexpr or templated
- `integral_constant.h`: template structs for integral constants.
All of the below are already in `<initializer_list>`
- `__config`: this is a bunch of c++ and specific defines macros
- `__config_site`: A bunch of defines for the LIBCPP abi namespace.
- `features.h`: Some more macros
- `features-time64.h: Macros for time bits
- `timesize.h`: Macros for time sizes
- `wordsize.h`: Macros for word sizes

Reply:
> however using explicit implementations for operator< and operator== to implement operator<= would call memcmp twice on the minimum size of the arrays being compared which I think is too expensive.

No it wouldn't: `x <= y` would be rewritten to `!(y < x)`.

Reply:
This isn't true for doubles right?

Reply:
`nan() < nan()` is false, and `nan() <= `nan()` is also false.

Reply:
What would be a way to measure the impact that that the previous commit versus the current commit:
> Do we really have to include this? This header includes dozens of other headers transitively, and including it here means adding all those file to every single translation unit in our entire codebase.

Is it something like compile time of a large binary?
Is there some kind of test we can add?